graylog2-web-interface/src/views/components/sidebar/FieldList.jsx

```diff
@@ -175,10 +179,29 @@ const FieldList = createReactClass({
           {this.showFieldsByLink('allreserved', 'all including reserved', 'This shows all fields, including reserved (gl2_*) fields.')} fields.
         </div>
         <hr />
-        {this._renderFieldList({ fields, allFields, showFieldsBy })}
+        {this._renderFieldList({ activeQueryFields, allFields, showFieldsBy })}
       </div>
     );
   },
 });
 
-export default FieldList;
+const FieldListWithContext = (props) => (
+  <FieldTypesContext.Consumer>
+    {({ all, queryFields } = {}) => {
+      const { viewMetadata: { activeQuery } } = props;
+      const activeQueryFields = queryFields?.get(activeQuery, all);
+      return <FieldList {...props} allFields={all} activeQueryFields={activeQueryFields} />;
+    }}
+  </FieldTypesContext.Consumer>
+);
+
+FieldListWithContext.propTypes = {
+  viewMetadata: PropTypes.object.isRequired,
+  listHeight: PropTypes.number,
+};
+
+FieldListWithContext.defaultProps = {
+  listHeight: 50,
+};
+
+export default connect(FieldListWithContext, { viewMetadata: ViewMetadataStore });
```

Comment:
Because we need `viewMetadata` for this component independently of the fields, I removed  the `ViewMetadataStore` connect from the `ExtendedSearchPage` `FieldList` and implemented it as a part of the component.

Reply:
The `ViewMetadataStore` is probably the next target that could be provided with a context instead.